okta/resource_okta_app_oauth_api_scope.go

```diff
@@ -0,0 +1,242 @@
+	result := make([]*okta.OAuth2ScopeConsentGrant, len(scopeIds))
+	for i, scopeId := range scopeIds {
+		result[i] = newOAuthApiScope(scopeId, issuer)
+	}
+	return result
+}
+
+// Fetches current granted application scopes and returns a map with names and IDs.
+func getOAuthApiScopeIdMap(ctx context.Context, d *schema.ResourceData, m interface{}) (map[string]string, error) {
+	result := make(map[string]string)
+	currentScopes, resp, err := getOktaClientFromMetadata(m).Application.ListScopeConsentGrants(ctx, d.Get("app_id").(string), nil)
+	if err := suppressErrorOn404(resp, err); err != nil {
+		return nil, fmt.Errorf("failed to get application scope consent grants: %v", err)
+	}
+	for _, currentScope := range currentScopes {
+		result[currentScope.ScopeId] = currentScope.Id
+	}
+	return result, nil
+}
+
+// set resource schema from a list scopes
+func setOAuthApiScopes(d *schema.ResourceData, to []*okta.OAuth2ScopeConsentGrant) error {
+	scopes := make([]string, len(to))
+	for i, scope := range to {
+		scopes[i] = scope.ScopeId
+	}
+	d.SetId(d.Get("app_id").(string))
+	_ = d.Set("issuer", d.Get("issuer").(string))
+	_ = d.Set("scopes", scopes)
+	return nil
+}
+
+// Grant a list of scopes to an OAuth application. For convenience this function takes a list of OAuth2ScopeConsentGrant structs.
+func grantOAuthApiScopes(ctx context.Context, d *schema.ResourceData, m interface{}, scopeGrants []*okta.OAuth2ScopeConsentGrant) error {
+	for _, scopeGrant := range scopeGrants {
+		_, _, err := getOktaClientFromMetadata(m).Application.GrantConsentToScope(ctx, d.Get("app_id").(string), *scopeGrant)
+		if err != nil {
+			return fmt.Errorf("failed to grant application api scope: %v", err)
+		}
+	}
+	return nil
+}
+
+// Revoke a list of scopes from an OAuth application. The scope ID is needed for a revoke.
+func revokeOAuthApiScope(ctx context.Context, d *schema.ResourceData, m interface{}, ids []string) error {
+	for _, id := range ids {
+		resp, err := getOktaClientFromMetadata(m).Application.RevokeScopeConsentGrant(ctx, d.Get("app_id").(string), id)
+		if err := suppressErrorOn404(resp, err); err != nil {
+			return fmt.Errorf("failed to revoke application api scope: %v", err)
+		}
+	}
+	return nil
+}
+
+// Diff function to identify which scope needs to be added or removed to the application
+func getOAuthApiScopeUpdateLists(d *schema.ResourceData, from []*okta.OAuth2ScopeConsentGrant) ([]string, []string) {
+	grantList := make([]string, 0)
+	revokeList := make([]string, 0)
+	desiredScopes := make([]string, 0)
+	currentScopes := make([]string, 0)
+
+	// cast list of interface{} to strings
+	for _, scope := range d.Get("scopes").([]interface{}) {
+		desiredScopes = append(desiredScopes, scope.(string))
+	}
+
+	// extract scope list form []okta.OAuth2ScopeConsentGrant
+	for _, currentScope := range from {
+		currentScopes = append(currentScopes, currentScope.ScopeId)
+	}
+
+	// find scopes that should not be there
```

Comment:
As an advice, you can use `splitTargets` (okta/resource_okta_admin_role_targets.go:183) func here. Yes, the naming is confusing, so feel free to rename params.

Reply:
Ah good catch, we have the same code already. I Will change that!